src/components/input/input.js

```diff
@@ -629,12 +631,28 @@ function mdSelectOnFocusDirective() {
     element.on('focus', onFocus);
 
     scope.$on('$destroy', function() {
-      element.off('focus', onFocus);
+      element
+        .off('focus', onFocus)
+        .off('mouseup', preventMouseUp);
     });
 
     function onFocus() {
-      // Use HTMLInputElement#select to fix firefox select issues
-      element[0].select();
+      // Use HTMLInputElement#select to fix firefox select issues.
+      // The debounce is here for Edge's sake, otherwise the selection doesn't work.
+      $timeout(function() {
+        element[0].select();
+      }, 1, false);
+
+      // Prevents the default action of the first `mouseup` after a focus.
+      // This is necessary, because browsers fire a `mouseup` right after the element
+      // has been focused. In some browsers (Firefox in particular) this can clear the
+      // selection. There are examples of the problem in issue #7487.
+      element.on('mouseup', preventMouseUp);
```

Comment:
Wouldn't it make more sense to keep the `mouseup` event registered and just change a state in the directive?.